### ISSUE TITLE
error in smart contract when executing a swap

### ISSUE THREAD

**Issue by cryptoB0T** [fun-xyz/funkit-react#23]
trying to get some clues into what's going wrong here. am able to create a wallet and stake, amongst other things; but can't get  swap to work on Goerli. getting the below error. 

- should i dig into the smart contract to look for what's going wrong? 
- do you have a demo version of this working? 
- will this only work on mainnet? 
- is there a limitation in certain API keys? 

here's the code: 
```
    const swapOp = await funWallet.swap(auth, await auth.getAddress(), {
      tokenIn: 'eth',
      tokenOut: 'usdc',
      inAmount: 0.1,
    });
```

in the console:
```
getCallError.js:10 Uncaught (in promise) CallExecutionError: Execution reverted with reason: Unexpected error.

Raw Call Arguments:
  to:    0x61fFE014bA17989E743c5F6cB21bF9697530B21e
  data:  0xc6a5026a000000000000000000000000b4fbf271143f4fbf7b91a5ded31805e42b2208d60000000000000000000000004df8d8aa018cabb7c1194e6df064961ea3572c48000000000000000000000000000000000000000000000000016345785d8a00000000000000000000000000000000000000000000000000000000000000000bb80000000000000000000000000000000000000000000000000000000000000000

Details: execution reverted: Unexpected error
Version: viem@0.3.50
    at getCallError (webpack-internal:///(app-pages-browser)/./node_modules/viem/dist/cjs/utils/errors/getCallError.js:10:12)
    at call (webpack-internal:///(app-pages-browser)/./node_modules/viem/dist/cjs/actions/public/call.js:73:50)
    at async SwapToken.getOutputQuote (webpack-internal:///(app-pages-browser)/./node_modules/@funkit/core/dist/src/utils/SwapUtils.js:46:37)
    at async SwapToken.createTrade (webpack-internal:///(app-pages-browser)/./node_modules/@funkit/core/dist/src/utils/SwapUtils.js:99:27)
    at async swapExec (webpack-internal:///(app-pages-browser)/./node_modules/@funkit/core/dist/src/utils/SwapUtils.js:152:47)
    at async uniswapV3SwapTransactionParams (webpack-internal:///(app-pages-browser)/./node_modules/@funkit/core/dist/src/actions/Swap.js:120:30)
    at async FunWallet.swap (webpack-internal:///(app-pages-browser)/./node_modules/@funkit/core/dist/src/actions/FirstClassActions.js:44:33)
    at async swapEth (webpack-internal:///(app-pages-browser)/./src/app/(app)/components/BuyComponent.tsx:31:24)
```

**Comment by cryptoB0T** [comment]
fyi this issue is also reproducible on the codesandbox 
https://codesandbox.io/p/github/TheFunGroup/Action-Web-Demo/main?file=/src/initializeBasicFunWallet.js

when you click to swap, you should see this error
```
ERROR
Execution reverted with reason: Unexpected error.

Raw Call Arguments:
  to:    0x61fFE014bA17989E743c5F6cB21bF9697530B21e
  data:  0xc6a5026a000000000000000000000000b4fbf271143f4fbf7b91a5ded31805e42b2208d60000000000000000000000004df8d8aa018cabb7c1194e6df064961ea3572c4800000000000000000000000000000000000000000000000000038d7ea4c680000000000000000000000000000000000000000000000000000000000000000bb80000000000000000000000000000000000000000000000000000000000000000

Details: execution reverted: Unexpected error
Version: viem@0.3.50
CallExecutionError: Execution reverted with reason: Unexpected error.

Raw Call Arguments:
  to:    0x61fFE014bA17989E743c5F6cB21bF9697530B21e
  data:  0xc6a5026a000000000000000000000000b4fbf271143f4fbf7b91a5ded31805e42b2208d60000000000000000000000004df8d8aa018cabb7c1194e6df064961ea3572c4800000000000000000000000000000000000000000000000000038d7ea4c680000000000000000000000000000000000000000000000000000000000000000bb80000000000000000000000000000000000000000000000000000000000000000

Details: execution reverted: Unexpected error
Version: viem@0.3.50
    at getCallError (https://43rv99-3000.csb.app/static/js/bundle.js:195780:10)
    at call (https://43rv99-3000.csb.app/static/js/bundle.js:184455:46)
    at async SwapToken.getOutputQuote (https://43rv99-3000.csb.app/static/js/bundle.js:30562:33)
    at async SwapToken.createTrade (https://43rv99-3000.csb.app/static/js/bundle.js:30611:23)
    at async swapExec (https://43rv99-3000.csb.app/static/js/bundle.js:30688:7)
    at async uniswapV3SwapTransactionParams (https://43rv99-3000.csb.app/static/js/bundle.js:26341:7)
    at async FunWallet.swap (https://43rv99-3000.csb.app/static/js/bundle.js:25558:27)
    at async swapEth (https://43rv99-3000.csb.app/static/js/bundle.js:150:16)
```

**Comment by AlbertSu123** [comment]
TLDR: The uniswap v3 pool that we use for testing on goerli ran out of liquidity. This issue is fixed!

`funwallet.swap()` uses a combination of 1inch and Uniswap V3 for swapping. For goerli, we are using uniswap v3. The token address for USDC is [0x4df8d8aa018cabb7c1194e6df064961ea3572c48](https://goerli.etherscan.io/address/0x4df8d8aa018cabb7c1194e6df064961ea3572c48). If it ever runs out of USDC or ETH, you can always add liquidity by minting the token in etherscan!

**Comment by cryptoB0T** [comment]
@AlbertSu123 this is great, thanks for the help! 

i was encountering this error below: 

```
sdk-core.esm.js:319 Uncaught (in promise) Error: Could not parse fraction
    at Function.tryParseFraction (webpack-internal:///(app-pages-browser)/./node_modules/@uniswap/sdk-core/dist/sdk-core.esm.js:374:11)
    at Percent.lessThan (webpack-internal:///(app-pages-browser)/./node_modules/@uniswap/sdk-core/dist/sdk-core.esm.js:405:32)
    at Trade.minimumAmountOut (webpack-internal:///(app-pages-browser)/./node_modules/@uniswap/v3-sdk/dist/v3-sdk.esm.js:3085:25)
    at eval (webpack-internal:///(app-pages-browser)/./node_modules/@uniswap/v3-sdk/dist/v3-sdk.esm.js:4035:28)
    at Array.reduce (<anonymous>)
    at Function.swapCallParameters (webpack-internal:///(app-pages-browser)/./node_modules/@uniswap/v3-sdk/dist/v3-sdk.esm.js:4034:33)
    at SwapToken.executeTrade (webpack-internal:///(app-pages-browser)/./node_modules/@funkit/core/dist/src/utils/SwapUtils.js:129:57)
    at swapExec (webpack-internal:///(app-pages-browser)/./node_modules/@funkit/core/dist/src/utils/SwapUtils.js:153:26)
    at async uniswapV3SwapTransactionParams (webpack-internal:///(app-pages-browser)/./node_modules/@funkit/core/dist/src/actions/Swap.js:120:30)
    at async FunWallet.swap (webpack-internal:///(app-pages-browser)/./node_modules/@funkit/core/dist/src/actions/FirstClassActions.js:44:33)
    at async swapEth (webpack-internal:///(app-pages-browser)/./src/app/(app)/demo/fun/social-web-fund-swap/SocialWebWrapperFundSwap.tsx:124:20)
```

this is the function that's triggering the error:
```
    const op = await funWallet.swap(auth, await auth.getUserId(), {
      tokenIn: 'eth',
      tokenOut: 'usdc',
      inAmount: 0.001,
    });
```

turned out that this issue was caused by a dependency mismatch

swap did not work when using `@uniswap/sdk-core@4.0.7`

but i when i switched to `@uniswap/sdk-core@4.0.6` then it worked 

i think you should probably update this line in funkit-core:
https://github.com/fun-xyz/funkit-core/blob/main/package.json#L37

i'd be curious to learn how exactly uniswap sdk introduced breaking changes in a patch release!

**Comment by Chazzzzzzz** [comment]
locked the uniswap SDK version in @funkit/core SDK

This PR should fix this. Thanks for pointing it out! https://github.com/fun-xyz/funkit-core/pull/366

will do a release soon

**Comment by Chazzzzzzz** [comment]
https://www.npmjs.com/package/@funkit/react

a new version is released and the issue should be fixed

**Comment by cryptoB0T** [comment]
hey @Chazzzzzzz, thanks for updating the pkg! can you share why fun is using `@uniswap/sdk-core ^3.0.1` instead of 4.0.6? was 4 causing other issues as well?
